contracts/Escrow.sol

```diff
@@ -193,22 +214,26 @@ contract Escrow {
 
         bulkPaid = false;
 
-        uint256 aggregatedBulkAmount = 0;
-        for (uint256 i; i < _amounts.length; i++) {
-            aggregatedBulkAmount += _amounts[i];
+        uint256 aggregatedBulkPayout = 0;
+        for (uint256 i; i < _payouts.length; i++) {
+            aggregatedBulkPayout += _payouts[i];
         }
 
-        if (balance < aggregatedBulkAmount) {
+        if (balance < aggregatedBulkPayout) {
             return bulkPaid;
         }
 
-        finalResultsUrl = _url;
-        finalResultsHash = _hash;
+        bool writeOnchain = bytes(_hash).length != 0 || bytes(_url).length != 0;
+        if (writeOnchain) {
+          // Be sure they are both zero if one of them is
+          reputationOracleResultsUrl = _url;
```

Comment:
Doesn't `_url` come from the contract fields? Is there a possibility that `_url` might not be empty from last chunk's payments? One contract can facilitate the HMT payments of multiple chunks.